src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -1,81 +1,322 @@
-    this.setPristine()
+
     this.isLoading = true
-    this.sessionService
-      .signUp(this.email, this.password, this.first_name, this.last_name)
-      .subscribe((data) => {
-        this.isLoading = false
-        this.onSuccess(data)
-      }, (error) => {
-        this.hasError = true
-        switch (error.status) {
-          case 400:
-            if (error.data && error.data.error === 'invalid_password') {
-              this.signUpErrors.invalid_password = true
-              this.invalidPasswordMessage = error.data.description
-            } else {
-              this.signUpErrors[error.status] = true
+    this.submitting = false
+  }
+
+  $onDestroy () {
+    this.oktaSignInWidget.remove()
+  }
+
+  setUpSignUpWidget () {
+    const donorData = this.donorDetails ?? {}
+    this.currentStep = 1
+
+    this.oktaSignInWidget = new OktaSignIn({
+      ...this.sessionService.oktaSignInWidgetDefaultOptions,
+      assets: {
+        baseUrl: '/assets/okta-sign-in/'
+      },
+      flow: 'signup',
+      registration: {
+        parseSchema: (schema, onSuccess) => {
+          const step = this.currentStep || 1
+          // Split the form into multiple steps for better user experience.
+          // Retain the values entered by the user in each step.
+          // Pre-populate the form fields with existing user details.
+
+          const organizationNameField = this.$scope.accountType === 'organization'
+            ? [{
+                name: 'organizationName',
+                type: 'text',
+                'label-top': true,
+                label: this.organizationNameTxt,
+                required: true,
+                maxLength: 50,
+                value: this.$scope.organizationName ?? donorData['organization-name'] ?? ''
+              }]
+            : []
+
+          const steps = {
+            // Step 1: Name, email and account type
+            1: [
+              {
+                ...schema[0],
+                value: this.$scope.firstName ?? donorData.name?.['given-name'] ?? this.sessionService.session.first_name ?? ''
+              },
+              {
+                ...schema[1],
+                value: this.$scope.lastName ?? donorData.name?.['family-name'] ?? this.sessionService.session.last_name ?? ''
+              },
+              {
+                ...schema[2],
+                value: this.$scope.email ?? donorData.email ?? this.sessionService.session.email ?? ''
+              },
+              {
+                name: 'accountType',
+                type: 'select',
+                options: {
+                  household: this.giveAsIndividualTxt,
+                  organization: this.giveAsOrganizationTxt
+                },
+                'label-top': true,
+                label: 'Account Type',
+                required: true,
+                wide: true,
+                value: this.$scope.accountType ?? donorData['donor-type'] ?? 'household'
```

Comment:
Cortex uses capitalized `'Household'` for the `donor-type`.

```suggestion
                options: {
                  Household: this.giveAsIndividualTxt,
                  Organization: this.giveAsOrganizationTxt
                },
                'label-top': true,
                label: 'Account Type',
                required: true,
                wide: true,
                value: this.$scope.accountType ?? donorData['donor-type'] ?? 'Household'
```

Reply:
I spotted this, this morning. I will push my code up